core/java/iesi-core/src/main/java/io/metadew/iesi/component/http/HttpComponentDefinitionService.java

```diff
@@ -59,6 +81,24 @@ public HttpComponentDefinition convert(Component component) {
                         .map(componentParameterValue -> HttpQueryParameterDefinitionService.getInstance().convert(componentParameterValue))
                         .collect(Collectors.toList())
         );
-    }
+        UUID uuid = UUID.randomUUID();
+        HttpComponentDesignTrace httpComponentDesignTrace = new HttpComponentDesignTrace(
+               new ComponentDesignTraceKey(uuid),
+                actionExecution.getExecutionControl().getRunId(),
+                actionExecution.getExecutionControl().getProcessId(),
+                actionParameterName,
+                COMPONENT_TYPE,
+                httpComponentDefinition.getReferenceName(),
+                httpComponentDefinition.getDescription(),
+                httpComponentDefinition.getVersion(),
+                httpComponentDefinition.getHttpConnectionReferenceName(),
+                httpComponentDefinition.getType(),
+                httpComponentDefinition.getEndpoint(),
+                httpComponentDefinition.getHeaders().stream().map(headers -> convertHeaders(headers,uuid)).collect(Collectors.toList()),
+                httpComponentDefinition.getQueryParameters().stream().map(queries -> convertQueries(queries,uuid)).collect(Collectors.toList())
+        );
+        ComponentDesignTraceConfiguration.getInstance().insert(httpComponentDesignTrace);
```

Comment:
could this be exported to a dedicated class 'HttpComponentDesignTraceService' with method 'public HttpComponentDesignTrace convert(HttpComponentDefinition httpComponentDefinition, ActionExecution actionExecution, String actionParameterName);